pkg/collector/corechecks/containers/containerd/check.go

```diff
@@ -140,6 +155,56 @@ func (c *ContainerdCheck) runContainerdCustom(sender sender.Sender) error {
 	return nil
 }
 
+func toSnakeCase(s string) string {
+	snake := matchAllCap.ReplaceAllString(s, "${1}_${2}")
+	return strings.ToLower(snake)
+}
+
+func (c *ContainerdCheck) collectImagePullMetrics(sender sender.Sender) error {
+	openmetricsEndpoint := fmt.Sprintf("%s/v1/metrics", c.instance.OpenmetricsEndpoint)
+	resp, err := c.httpClient.Get(openmetricsEndpoint)
+	if err != nil {
+		return err
+	}
+
+	defer resp.Body.Close()
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return err
+	}
+
+	parsedMetrics, err := prometheus.ParseMetrics(body)
+	if err != nil {
+		return err
+	}
+
+	for _, sample := range parsedMetrics {
+		if sample == nil {
+			continue
+		}
+
+		metric := sample.Metric
+
+		grpcMethod, ok := metric["grpc_method"]
+		if !ok || grpcMethod != pullImageGrpcMethod {
+			continue
+		}
+
+		grpcCode, ok := metric["grpc_code"]
+		if !ok {
+			continue
+		}
+
+		metricName := fmt.Sprintf("containerd.image.pull.%s", toSnakeCase(string(grpcCode)))
+		metricTags := []string{fmt.Sprintf("grpc_service:%s", metric["grpc_service"])}
+
+		sender.Counter(metricName, float64(sample.Value), "", metricTags)
```

Comment:
In order to accurately report a count of events from a Prometheus monotonic counter, we need to use `MonotonicCount` instead.

```suggestion
		sender.MonotonicCount(metricName, float64(sample.Value), "", metricTags)
```